qiskit/providers/ibmq/__init__.py

```diff
@@ -93,8 +121,13 @@ def least_busy(backends: List[BaseBackend]) -> BaseBackend:
             does not have the ``pending_jobs`` attribute in its status.
     """
     try:
-        return min([b for b in backends if b.status().operational],
-                   key=lambda b: b.status().pending_jobs)
+        backends_to_filter = {}
+        for backend in backends:
+            backend_status = backend.status()
+            if backend_status.operational:
+                backends_to_filter[backend] = backend_status
+
+        return min(backends_to_filter.keys(), key=lambda b: backends_to_filter[b].pending_jobs)
```

Comment:
I changed it because `status()` was being called twice for every backend and this was cluttering the log. But, we're not going to log `/devices/<device_name>/status` so I'll change it back.